src/sodaspark/scan.py

```diff
@@ -297,3 +301,122 @@ def execute(
     scan = create_scan(scan_yml, soda_server_client=soda_server_client)
     scan.execute()
     return scan.scan_result
+
+def measurements_to_data_frame(measurements: list[Measurement]) -> DataFrame:
+    """
+    Convert measurements to a data frame.
+    Parameters
+    ----------
+    measurements: List[Measurement]
+        The measurements.
+    Returns
+    -------
+    out : DataFrame
+        The measurements as data frame.
+    """
+    schema_group_values = T.StructType(
+        [
+            T.StructField("group", T.StringType(), True),
+            T.StructField("value", T.StringType(), True),
+        ]
+    )
+    schema = T.StructType(
+        [
+            T.StructField("metric", T.StringType(), True),
+            T.StructField("columnName", T.StringType(), True),
+            T.StructField("value", T.StringType(), True),
+            T.StructField("groupValues", schema_group_values, True),
+        ]
+    )
+    spark_session = SparkSession.builder.getOrCreate()
+    out = spark_session.createDataFrame(
+        [measurement.to_dict() for measurement in measurements],
```

Comment:
you might not need the `to_dict` (and list comprehension). In the [tests](https://github.com/sodadata/soda-spark/blob/4b433169da297fbceb82e2e0ca9a3282a86a4a09/tests/test_scan.py#L173) we also convert a list of data classes to a data frame without conversion.

Reply:
I have updated this for measurement and scan_error. But for Test_Result I had to define the schema explicitly to get the all captured details into the Dataframe.